syntax = "proto3";

package qbrix.common;

message Context {
  string id = 1;
  repeated float vector = 2;
  map<string, string> metadata = 3;
}

// Arm representation
message Arm {
  string id = 1;
  string name = 2;
  int32 index = 3;
}

// Pool of arms
message Pool {
  string id = 1;
  string name = 2;
  repeated Arm arms = 3;
}

// Experiment configuration
message Experiment {
  string id = 1;
  string name = 2;
  string pool_id = 3;
  string protocol = 4;  // e.g., "thompson_sampling", "ucb", "epsilon_greedy"
  map<string, string> protocol_params = 5;
  bool enabled = 6;
}

// Health check
message HealthCheckRequest {}

message HealthCheckResponse {
  enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
  }
  ServingStatus status = 1;
}
